src/components/AppBar/Drawer.tsx

```diff
@@ -0,0 +1,103 @@
+import * as React from 'react';
+import Box from '@mui/material/Box';
+import Drawer from '@mui/material/Drawer';
+import List from '@mui/material/List';
+import Divider from '@mui/material/Divider';
+import ListItem from '@mui/material/ListItem';
+import ListItemButton from '@mui/material/ListItemButton';
+import ListItemIcon from '@mui/material/ListItemIcon';
+import ListItemText from '@mui/material/ListItemText';
+import IconButton from '@mui/material/IconButton';
+import SailingIcon from '@mui/icons-material/Sailing';
+import Avatar from '@mui/material/Avatar';
+import LogoutIcon from '@mui/icons-material/Logout';
+import SportsEsportsIcon from '@mui/icons-material/SportsEsports';
+import ForumIcon from '@mui/icons-material/Forum';
+
+export default () => {
+  const [isOpened, setIsOpened] = React.useState(false)
+
+  const toggleDrawer = (open: boolean) => (event: React.KeyboardEvent | React.MouseEvent) => {
+    if (
+      event.type === 'keydown'
+      && ((event as React.KeyboardEvent).key === 'Tab'
+      || (event as React.KeyboardEvent).key === 'Shift')
+    ) { return; }
+
+    setIsOpened(open)
+  }
+
+  const goToProfile = () => {
+    console.log('AAAAAAA To Profile')
+  }
+
+  const goToGame = () => {
+    console.log('AAAAAAA To Game')
+  }
+
+  const goToForum = () => {
+    console.log('AAAAAAA To Forum')
+  }
+
+  const logout = () => {
+    console.log('AAAAAAA Logout')
+  }
+
+  const list = () => (
+    <Box
+      sx={{ width: 200 }}
+      role='presentation'
+      onClick={toggleDrawer(false)}
```

Comment:
❗️
Это вызов функции, она возвращает `undefined`)
```suggestion
      onClick={() => toggleDrawer(false)}
```
Или
```suggestion
      onClick={closeDrawer}
```

Reply:
> `onClick={() => toggleDrawer(false)}`


а не будет ли создаваться новая функция каждый раз при отрисовке компонента?

Reply:
Будет, конечно, но сейчас у тебя в обработчике не функция)
Если ты указываешь функции в теле компонента и не мемоизируешь, то она точно так же создаётся на каждый рендер